### PR TITLE
fix(curriculum): Merge Sort algorithm step 10 requires spaces to pass

### LINKED ISSUE #55977: Merge Sort algorithm step 10 requires spaces to pass
### Describe the Issue

In this challenge if one doesn't use spaces around the euql sign, the tests fail

### Affected Page

https://www.freecodecamp.org/learn/scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/step-10

### Your code

```py
def merge_sort(array):
    middle_point = len(array) // 2
    left_part = array[:middle_point]
    right_part = array[middle_point:]
    merge_sort(left_part)
    merge_sort(right_part)

# User Editable Region


    left_array_index=0
    right_array_index=0
    sorted_index=0


# User Editable Region


```

### Expected behavior

I expect it should pass, instead it passes only if converted to

```py
    left_array_index = 0
    right_array_index = 0
    sorted_index = 0
```

### Screenshots

_No response_

### System

NA

### Additional context

This could be an excuse to convert the challenge to AST helpers, where spaces are not an issue.

### PR DESCRIPTION
**Pull request by subhadipjana866**
This pull request fixes the issue number #55977 by adding /s* in the regular expressions.  The issue is fixed.

Fix #55977

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55977

### CODE REVIEW

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ You should have a variable named `left_array_index` inside your `merge_sort` fun
     const merge_sort = __helpers.python.getDef("\n" + transformedCode, "merge_sort");
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /left_array_index\s*(?!=)/);
+    assert.match(function_body, /left_array_index\s*(?!=)\s*/);
```
Comment (lasjorg) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md:27]:
```suggestion
    assert.match(function_body, /left_array_index/);
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ You should have a variable named `right_array_index` inside your `merge_sort` fu
     const merge_sort = __helpers.python.getDef("\n" + transformedCode, "merge_sort");
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /right_array_index\s*(?!=)/);
+    assert.match(function_body, /right_array_index\s*(?!=)\s*/);
```
Comment (lasjorg) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md:55]:
```suggestion
    assert.match(function_body, /right_array_index/);
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md [hunk 3]
```diff
@@ -81,7 +81,7 @@ You should have a variable named `sorted_index` inside your `merge_sort` functio
     const merge_sort = __helpers.python.getDef("\n" + transformedCode, "merge_sort");
     const { function_body } = merge_sort;
 
-    assert.match(function_body, /sorted_index\s*(?!=)/);
+    assert.match(function_body, /sorted_index\s*(?!=)\s*/);
```
Comment (lasjorg) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-data-structures-by-building-the-merge-sort-algorithm/656680b0fc79f2c38a34d90e.md:84]:
```suggestion
    assert.match(function_body, /sorted_index/);
```

**Review by lasjorg — CHANGES_REQUESTED** [review]
I know you didn't write it, but I don't understand the negative look ahead. It makes the optional space required (as far as I can tell).

Considering identifiers without an assignment is valid syntax, but an assignment without a value isn't, wouldn't it make more sense to just have the first regex for each identifier only look for the identifier?

**Review by lasjorg — APPROVED** [review]
LGTM

**Review by moT01 — APPROVED** [review]
LGTM @subhadipjana866 🎉 Congratulations on your first contribution to the freeCodeCamp repo 🎉

**Pull request merged**